systemtest/src/test/java/io/strimzi/systemtest/report/NodePoolsClusterReportST.java

```diff
@@ -0,0 +1,488 @@
+            "kafkabridges.kafka.strimzi.io.yaml",
+            "kafkaconnects.kafka.strimzi.io.yaml",
+            "kafkamirrormaker2s.kafka.strimzi.io.yaml",
+            "kafkanodepools.kafka.strimzi.io.yaml",
+            "kafkarebalances.kafka.strimzi.io.yaml",
+            "kafkas.kafka.strimzi.io.yaml",
+            "kafkatopics.kafka.strimzi.io.yaml",
+            "kafkausers.kafka.strimzi.io.yaml",
+            "strimzipodsets.core.strimzi.io.yaml"
+        )) {
+            assertValidYamls(outPath + "/reports/customresourcedefinitions", CustomResourceDefinition.class, s, 1);
+        }
+    }
+
+    private void assertValidCustomDeployments(String outPath, String clusterName) throws IOException {
+        for (String s : Arrays.asList(
+            "my-bridge-bridge.yaml",
+            clusterName + "-cruise-control.yaml",
+            clusterName + "-entity-operator.yaml",
+            "strimzi-cluster-operator.yaml"
+        )) {
+            assertValidYamls(outPath + "/reports/deployments", Deployment.class, s, 1);
+        }
+    }
+
+    private void assertValidNetworkPolicies(String outPath, String clusterName) throws IOException {
+        for (String s : Arrays.asList(
+            clusterName + "-entity-operator.yaml",
+            clusterName + "-network-policy-cruise-control.yaml",
+            clusterName + "-network-policy-kafka.yaml",
+            clusterName + "-network-policy-zookeeper.yaml",
+            "my-mm2-mirrormaker2.yaml"
+        )) {
+            assertValidYamls(outPath + "/reports/networkpolicies", NetworkPolicy.class, s, 1);
+        }
+    }
+
+    private void assertValidPodDisruptionBudgets(String outPath, String clusterName) throws IOException {
+        for (String s : Arrays.asList(
+            "my-bridge-bridge.yaml",
+            clusterName + "-kafka.yaml",
+            clusterName + "-zookeeper.yaml",
+            "my-connect-connect.yaml",
+            "my-mm2-mirrormaker2.yaml"
+        )) {
+            assertValidYamls(outPath + "/reports/poddisruptionbudgets", PodDisruptionBudget.class, s, 1);
+        }
+    }
+
+    private void assertValidPods(String outPath, String clusterName) throws IOException {
+        for (String s : Arrays.asList(
+            clusterName + "-pool-a-0.yaml",
+            clusterName + "-pool-a-1.yaml",
+            clusterName + "-pool-b-2.yaml",
+            clusterName + "-zookeeper-0.yaml",
+            clusterName + "-zookeeper-1.yaml",
+            clusterName + "-zookeeper-2.yaml",
+            "my-connect-connect-0.yaml",
+            "my-mm2-mirrormaker2-0.yaml"
+        )) {
+            assertValidYamls(outPath + "/reports/pods", Pod.class, s, 1);
+        }
+        assertValidYamls(outPath + "/reports/pods", Pod.class, "my-bridge-bridge", 1);
+        assertValidYamls(outPath + "/reports/pods", Pod.class, clusterName + "-cruise-control", 1);
+        assertValidYamls(outPath + "/reports/pods", Pod.class, clusterName + "-entity-operator", 1);
```

Comment:
Above three could be in the for loop?

Reply:
Nope, as they leverage the prefix logic here (no stable identities in standard deploy).